src/sql/workbench/services/notebook/common/sql/sqlSessionManager.ts

```diff
@@ -174,6 +176,7 @@ class SqlKernel extends Disposable implements nb.IKernel {
 	) {
 		super();
 		this.initMagics();
+		this.setConnectionPath();
```

Comment:
is it possible to go to all the places that consumes this and make sure the path passed in is a valid URI.
the risk of doing this is you are modifying the passed value, but the caller might rely on the value for something else.

Reply:
Thanks. I haven't seen any ill effects here,  and validated that we only set/read this value in the SqlKernel, as the connection that we use in conjunction with the queryRunner is only referenced here. Other connections are not impacted.

Tested the following:
- Queries succeed both for SQL and Spark kernels
- IntelliSense works for one + multiple cells. Switching between cells results in correct IntelliSense
- Connections get cleaned up when notebook is closed (both that query runner uses + IntelliSense)